support-metrics-client/src/main/java/io/confluent/support/metrics/MetricsReporter.java

```diff
@@ -106,7 +106,7 @@ public MetricsReporter(KafkaServer server,
     customerId = SupportConfig.getCustomerId(serverConfiguration);
     TimeUtils time = new TimeUtils();
     if (SupportConfig.isAnonymousUser(customerId)) {
-      CollectorFactory factory = new CollectorFactory(CollectorType.BASIC, time, null, null, null);
+      CollectorFactory factory = new CollectorFactory(CollectorType.BASIC, time, server, serverConfiguration, serverRuntime);
```

Comment:
I thought we omitted these in this case because we shouldn't be collecting this info for anonymous (non-customer) users?